### PR TITLE
perf(kafka): update events_plugin_ingestion partitioning and consumption

### PR DESCRIPTION
**Pull request by yakkomajuri**
- Updates kafka partition key to be hash(team_id, distinct_id) so we can see how it performs, start monitoring it and do some analysis on the batches
- Add config for `partitionsConsumedConcurrently` in the plugin server. This should help us with performance so let's see how it does. Currently set to 3 but will effectively be following [docs](https://kafka.js.org/docs/consuming) advice here:

> "Depending on whether or not your workload is CPU bound, it may also not benefit you to set it to a higher number than the number of logical CPU cores. A recommendation is to start with a low number and measure if increasing leads to higher throughput."

### CODE REVIEW

File: task-definition.plugins.json [hunk 1]
```diff
@@ -91,6 +91,10 @@
                 {
                     "name": "ASYNC_MIGRATIONS_DISABLE_AUTO_ROLLBACK",
                     "value": "1"
+                },
+                {
+                    "name": "KAFKA_PARTITIONS_CONSUMED_CONCURRENTLY",
+                    "value": "3"
```
Comment (macobo) [inline task-definition.plugins.json:97]: Can we deploy this separately from partitioning? This way when _something_ blows up we have an obvious revert target.

File: posthog/api/capture.py [hunk 2]
```diff
@@ -306,4 +307,5 @@ def capture_internal(event, distinct_id, ip, site_url, now, sent_at, team_id, ev
         sent_at=sent_at,
         event_uuid=event_uuid,
     )
-    log_event(parsed_event, event["event"])
+    partition_key = hashlib.sha256(f"{team_id}:{distinct_id}".encode()).hexdigest()
```
Comment (macobo) [inline posthog/api/capture.py:310]: Sanity check: Will plugin server be able to hash this identically way if needed in the future?

File: task-definition.plugins.json (hunk 1 above)
Comment (yakkomajuri) [reply task-definition.plugins.json:97]: yup, will split out - was thinking of doing this proactively anyway after the review

File: posthog/api/capture.py (hunk 2 above)
Comment (yakkomajuri) [reply posthog/api/capture.py:310]: Yup, we have team id and distinct id there and we can get sha256 digests the same way

actually - this reminds me we do produce to kafka from the plugin server already, will make sure we hash there too

**Review by macobo — APPROVED** [review]
Approved pending splitting / doing the same in plugin-server.

File: plugin-server/src/worker/vm/extensions/posthog.ts [hunk 3]
```diff
@@ -32,12 +32,16 @@ export function createPosthog(server: Hub, pluginConfig: PluginConfig): DummyPos
             if (!server.kafkaProducer) {
                 throw new Error('kafkaProducer not configured!')
             }
-            // ignore the promise, run in the background just like with celery
+
+            const partitionKeyHash = crypto.createHash('sha256')
+            partitionKeyHash.update(`${data.team_id}:${data.distinct_id}`)
+            const partitionKey = partitionKeyHash.digest('hex')
```
Comment (yakkomajuri) [inline plugin-server/src/worker/vm/extensions/posthog.ts:38]: @macobo

File: posthog/api/capture.py (hunk 2 above)
Comment (yakkomajuri) [reply posthog/api/capture.py:310]: tagged you in a comment about this

File: plugin-server/src/worker/vm/extensions/posthog.ts (hunk 3 above)
Comment (macobo) [reply plugin-server/src/worker/vm/extensions/posthog.ts:38]: perfect, thank you!

**Pull request merged**